pombola/hansard/management/commands/hansard_check_for_new_sources.py

```diff
@@ -95,42 +108,67 @@ def process_url(self, list_page, url, verbose):
                 if verbose:
                     message = "{0}: Skipping page with name: {1}"
                     print message.format(list_page, name)
+                get_next_page = False
             else:
                 if verbose:
                     message = "{0} Trying to add page with name {1} as a new source"
                     print message.format(list_page, name)
+                if next_url:
+                    get_next_page = True
+
+                self.extract_source_from_html(href, name, list_page, links, link_sections, idx, base_url)
+
+        # Scrape the next page if the last item on the page was added
+        # as a new source and the next_page_button contains a link
+        # (assumes that Hansard docs are being listed - and consumed -
+        # in date order, newest first)
+        #
+        # Hopefully requiring a second page of data will be a rarity;
+        # potentially fragile if the connection is lost before traversing
+        # all necessary pages
+        if get_next_page:
```

Comment:
Rather than have some of the next page logic here, and making a recursive call, I'd be tempted to write a generator, `get_index_pages`, say, that takes a URL like `http://www.parliament.go.ke/index.php/the-senate/house-business/hansard` and `yield`s a soup and base_url for each index page, so that all the page iteration logic is in a separate function (except for when you decided to break out of the iteration). It'd be fine like this too though.